### PR TITLE
Splitting seq.fs

### PR DESCRIPTION
**Pull request by manofstick**
As proposed in [this comment](https://github.com/Microsoft/visualfsharp/pull/2587#issuecomment-285929483) this should be a prelude to that #2587 - if this PR is merged and #2587 rebased then it should be provide a simpler set of diffs.

No functionality or change in logic is modified with this PR, it is just a split of the code that is from the `seq.fs(i)` files.

This has some benefit even without #2587 as it provides a physical separation of functionality that is used by the compiler for implementation of seq computational expressions that is shared with the Seq module.

### CODE REVIEW

**Comment by KevinRansom** [comment]
@jack-pappas  --- in principle I like this split regardless of the outcome of iseq.

It will take a while to review it I'm afraid.

Kevin

File: src/fsharp/FSharp.Core/seq.fs [hunk 1]
```diff
@@ -13,49 +13,9 @@ namespace Microsoft.FSharp.Collections
...
-                  (alreadyFinished() : obj)
-              member x.MoveNext() =
-                  if not started then started <- true
-                  false
-              member x.Reset() = noReset()
-          interface System.IDisposable with
-                member x.Dispose() = ()
-
-      let Empty<'T> () = (new EmptyEnumerator<'T>() :> IEnumerator<'T>)
+    module Internal =
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/seq.fs]: This is a bit idiosyncratic ... 

perhaps if you renamed the namespace containing SeqCore to Microsoft.FSharp.Collections.SeqCore and opened Microsoft.FSharp.Collections.SeqCore.IEnumerable?  Then maybe not jam in a new module.

Comment (manofstick) [inline src/fsharp/FSharp.Core/seq.fs]: Just trying to minimize lines code changed; maybe module naming could at least serve as as a temporary names until this is consumed?

Comment (dsyme) [inline src/fsharp/FSharp.Core/seq.fs]: I'm OK with this as a transitional step in rearranging the code with minimal diff

**Pull request merged**